### PR TITLE
Bluetooth: controller: Fix missing auxiliary advertising start

### PR DESCRIPTION
**Pull request by cvinayak**
Fix missing auxiliary channel advertising start when
starting Periodic Advertising.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 1]
```diff
@@ -195,66 +197,131 @@ uint8_t ll_adv_sync_enable(uint8_t handle, uint8_t enable)
...
+		}
+
+		sync->is_started = 0U;
+	}
+
+	if (!enable) {
+		sync->is_enabled = 0U;
+	}
+
+	return err_val;
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:324]: Please move this into a separate function. I do not think the potential savings warrant the use of goto in this case.

**Review by wopu-ot — CHANGES_REQUESTED** [review]

**Review by andrzej-kaczmarek — APPROVED** [review]
lgtm, just some nitpick

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 2]
```diff
@@ -812,17 +810,17 @@ uint32_t ull_adv_aux_evt_init(struct ll_adv_aux_set *aux)
 }
 
 uint32_t ull_adv_aux_start(struct ll_adv_aux_set *aux, uint32_t ticks_anchor,
-			   uint32_t ticks_slot_overhead,
-			   uint32_t volatile *ret_cb)
+			   uint32_t ticks_slot_overhead)
 {
+	uint32_t volatile ret_cb;
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:815]: agreed. But will keep it for the sake of conveying that the value is updated in ISR or another thread.

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_adv_sync.c]: Moved it to an inline function.

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 3]
```diff
@@ -195,63 +198,93 @@ uint8_t ll_adv_sync_enable(uint8_t handle, uint8_t enable)
...
+			aux->interval = adv->interval +
+					(HAL_TICKER_TICKS_TO_US(
+						ULL_ADV_RANDOM_DELAY) /	625U);
+
+			ret = ull_adv_aux_start(aux, ticks_anchor_aux,
+						ticks_slot_overhead_aux);
+			if (ret) {
+				err = sync_remove(sync, adv, 1U,
+					BT_HCI_ERR_INSUFFICIENT_RESOURCES);
+				return err;
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:279]:
```suggestion
				sync_remove(sync, adv, 1U);
				return BT_HCI_ERR_INSUFFICIENT_RESOURCES;
```

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 4]
```diff
@@ -401,6 +439,50 @@ static inline uint8_t sync_stop(struct ll_adv_sync_set *sync)
...
+
+	/* Remove sync_info from auxiliary PDU */
+	err = ull_adv_aux_hdr_set_clear(adv, 0,
+					ULL_ADV_PDU_HDR_FIELD_SYNC_INFO,
+					NULL, NULL, &pri_idx);
+	if (err) {
+		if (err_val) {
+			return err_val;
+		}
+
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:457]:
```suggestion
```

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 5]
```diff
@@ -401,6 +439,50 @@ static inline uint8_t sync_stop(struct ll_adv_sync_set *sync)
...
+	if (sync->is_started) {
+		/* TODO: we removed sync info, but if sync_stop() fails, what do
+		 * we do?
+		 */
+		err = sync_stop(sync);
+		if (err) {
+			if (err_val) {
+				return err_val;
+			}
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:471]:
```suggestion
```

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 6]
```diff
@@ -401,6 +439,50 @@ static inline uint8_t sync_stop(struct ll_adv_sync_set *sync)
...
+		}
+
+		sync->is_started = 0U;
+	}
+
+	if (!enable) {
+		sync->is_enabled = 0U;
+	}
+
+	return err_val;
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:483]:
```suggestion
	return 0U;
```

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 7]
```diff
@@ -401,6 +439,50 @@ static inline uint8_t sync_stop(struct ll_adv_sync_set *sync)
 	return 0;
 }
 
+static inline uint8_t sync_remove(struct ll_adv_sync_set *sync,
+				  struct ll_adv_set *adv, uint8_t enable,
+				  uint8_t err_val)
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:444]:
```suggestion
static inline uint8_t sync_remove(struct ll_adv_sync_set *sync,
				  struct ll_adv_set *adv, uint8_t enable)
```

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 8]
```diff
@@ -42,7 +42,9 @@ static int init_reset(void);
 static inline struct ll_adv_sync_set *sync_acquire(void);
 static inline void sync_release(struct ll_adv_sync_set *sync);
 static inline uint16_t sync_handle_get(struct ll_adv_sync_set *sync);
-static inline uint8_t sync_stop(struct ll_adv_sync_set *sync);
+static inline uint8_t sync_remove(struct ll_adv_sync_set *sync,
+				  struct ll_adv_set *adv, uint8_t enable,
+				  uint8_t err_val);
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:47]:
```suggestion
static inline uint8_t sync_remove(struct ll_adv_sync_set *sync,
				  struct ll_adv_set *adv, uint8_t enable);
```

**Review by wopu-ot — CHANGES_REQUESTED** [review]
I believe the error value handling in sync_remove could and should be simplified (see suggestions).

**Review by wopu-ot — APPROVED** [review]
LGTM

**Review by kruithofa — APPROVED** [review]

**Pull request merged**